### PR TITLE
Application settings service

### PR DESCRIPTION
**Pull request by sivakumargollu**
Simple design for app-setting service  with classes and objects
Contains service endpoints to manage application-settings and authentication information. It also contains logger config  endpoints.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
-tmp/
-test-result
-server.pid
-*.eml
-/dist/
-.cache
+
+target/
+*.ipr
+*.iws
```
Comment (eklavya) [inline .gitignore:4]: This doesn't make any sense. Use what others are using.

File: src/main/scala/com/activegrid/models/LogConfigUpdater.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import org.apache.log4j._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class LogConfigUpdater(implicit ex: ExecutionContext) {
+
+  val ROOT: String = "root"
+
+  def getLogger(logger: String): Logger = {
```
Comment (eklavya) [inline src/main/scala/com/activegrid/models/LogConfigUpdater.scala]: What is this?

File: src/main/scala/com/activegrid/models/LogConfigUpdater.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    logger match {
+      case ROOT => getLogger(ROOT)
+      case _ => getLogger(logger)
+    }
+  }
+
+  def getLogLevel2(logger: String): Future[String] = Future {
```
Comment (eklavya) [inline src/main/scala/com/activegrid/models/LogConfigUpdater.scala]: Why do we need all this?

File: src/main/scala/com/activegrid/neo4j/AppSettingsNeo4jWrapper.scala [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    }
+  }
+
+  def updateSettings(settingsMap: Map[String, String], relationName: String): String = {
+    withTx {
+      neo => {
+        var genaralSettings = Map.empty[String, String];
+        withTx { neo =>
+          val settingNodes = getAllNodesWithLabel(lables.get("AS").toString)(neo)
+          for (n <- settingNodes) {
```
Comment (eklavya) [inline src/main/scala/com/activegrid/neo4j/AppSettingsNeo4jWrapper.scala]: What is this?

File: src/main/scala/com/activegrid/neo4j/AppSettingsNeo4jWrapper.scala [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    }
+  }
+
+  def deleteSetting(settingsMap: Map[String, String], relationName: String): String = {
+    withTx {
+      neo => {
+        var genaralSettings = Map.empty[String, String];
+        withTx { neo =>
+          val settingNodes = getAllNodesWithLabel(lables.get("AS").toString)(neo)
+          for (n <- settingNodes) {
```
Comment (eklavya) [inline src/main/scala/com/activegrid/neo4j/AppSettingsNeo4jWrapper.scala]: What is this?

**Review by eklavya — CHANGES_REQUESTED** [review]

File: src/main/scala/com/activegrid/neo4j/AppSettingsNeo4jWrapper.scala (hunk 5 above)
Comment (sivakumargollu) [reply src/main/scala/com/activegrid/neo4j/AppSettingsNeo4jWrapper.scala]: "AS" is a key in labels map. It associates with value AppSettings.

File: src/main/scala/com/activegrid/models/LogConfigUpdater.scala (hunk 3 above)
Comment (sivakumargollu) [reply src/main/scala/com/activegrid/models/LogConfigUpdater.scala]: We have end points that return back  LOGGER related information. We couldn't find any use case but might required by other services. Now corrected the method definition.

File: src/main/scala/com/activegrid/models/LogConfigUpdater.scala (hunk 2 above)
Comment (sivakumargollu) [reply src/main/scala/com/activegrid/models/LogConfigUpdater.scala]: Method definition corrected

File: .gitignore (hunk 1 above)
Comment (sivakumargollu) [reply .gitignore:4]: .ipr and .iws are the intellej auto-generated files which contains project files related information. To make it simple and clean, except source files remaining all are excluded and also unused statements from .gitigore removed.

File: src/main/scala/com/activegrid/neo4j/AppSettingsNeo4jWrapper.scala [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+              }
+            }
+          }
+          AppSettings(genaralSettings, authSettings);
+        }
+      }
+    }
+  }
+
+  def updateSettings(settingsMap: Map[String, String], relationName: String): String = {
```
Comment (eklavya) [inline src/main/scala/com/activegrid/neo4j/AppSettingsNeo4jWrapper.scala]: Why are you creating a string of success and failure?

Comment (sivakumargollu) [reply src/main/scala/com/activegrid/neo4j/AppSettingsNeo4jWrapper.scala]: I had created string to   use it as STATUS that let the function caller know weather  operation succeeded or not. I changed this just committed code please review it

**Pull request merged**